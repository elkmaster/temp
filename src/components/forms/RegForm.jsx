import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
// TODO: validation
export const RegForm = ({ handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    <Field
      name="first_name"
      component="input"
      type="text"
      placeholder="First name"
    />
    <Field
      name="last_name"
      component="input"
      type="text"
      placeholder="Last name"
    />
    <Field
      name="email"
      component="input"
      type="email"
      placeholder="email"
    />
    <Field
      name="password"
      component="input"
      type="password"
      placeholder="pass"
    />
    <button type="submit">Sign Up</button>
  </form>
);

RegForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default reduxForm({
  form: 'regForm', // a unique identifier for this form
  touchOnBlur: false,
})(RegForm);
