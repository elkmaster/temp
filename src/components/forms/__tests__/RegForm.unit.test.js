import React from 'react';
import { shallow,  } from 'enzyme';
import faker from 'faker';

import { RegForm } from '../RegForm';

const setup = (propsOverrides) => {
  const props = Object.assign({
      handleSubmit: jest.fn(),
    },
    propsOverrides);

  const wrapper = shallow(<RegForm {...props} />);

  return {
    props,
    wrapper,
  };
};

describe('RegForm', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  //TODO: check if there is more optimal way to query field types????
  it('should has correct input fields', () => {
    const { wrapper } = setup();
    expect(wrapper.findWhere(n => n.props().name == 'first_name' && n.props().type == 'text')).toHaveLength(1);
    expect(wrapper.findWhere(n => n.props().name == 'last_name' && n.props().type == 'text')).toHaveLength(1);
    expect(wrapper.findWhere(n => n.props().name == 'email' && n.props().type == 'email')).toHaveLength(1);
    expect(wrapper.findWhere(n => n.props().name == 'password' && n.props().type == 'password')).toHaveLength(1);

  });

  describe('form submit', () => {
    it('should call handleSubmit on submit', () => {
      const { wrapper, props } = setup();
      wrapper.simulate('submit');
      expect(props.handleSubmit).toBeCalled();
    });

    it('should call handleSubmit with correct params', () => {
      const { wrapper, props } = setup();
      const payload = {
        values: {
          first_name: faker.name.firstName(),
          last_name: faker.name.lastName(),
          email: faker.internet.email(),
          password: faker.internet.password(),
        },
      };
      wrapper.simulate('submit', payload);

      expect(props.handleSubmit).toBeCalledWith(payload);
    });
  });

});
