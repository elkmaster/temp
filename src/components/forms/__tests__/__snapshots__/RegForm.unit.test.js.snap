// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Field
    component="input"
    name="first_name"
    placeholder="First name"
    type="text"
  />
  <Field
    component="input"
    name="last_name"
    placeholder="Last name"
    type="text"
  />
  <Field
    component="input"
    name="email"
    placeholder="email"
    type="email"
  />
  <Field
    component="input"
    name="password"
    placeholder="pass"
    type="password"
  />
  <button
    type="submit"
  >
    Sign Up
  </button>
</form>
`;
